docs/proposals/proposal-workflow-authentication.md

```diff
@@ -0,0 +1,69 @@
+# Proposal for Workflow Authentication
+
+## Proposal summary
+
+**For workflow-to-workflow-service authentication, use:**
+
+* **for jobs: one-time per-job tokens**, i.e. a single one-time token for
+  every job;
+* **for the user/interactive CLI: timed tokens**.
+
+Per-job tokens would require one of the following (or another alternative):
+
+  * another file per job, which is not ideal as it adds to the file count;
+  * a shared combined file containing tokens for all jobs per workflow, also
+    not ideal as it means that jobs would needlessly know about other jobs'
+    tokens;
+  * extending an existing job file such as the 'job.status' so that these files
+    also carry the token information for each job, & preventing that
+    information from being displayed in Cylc Review.
+
+Decisions on implementation details, such as modules & standards to use,
+await the go-ahead on an approach, the above one or otherwise.
+
+
+## Further information
+
+### Context
+
+Cylc 8 discussions have converged (as described briefly
+[here](cylc-8-architecture#command-line-interface) &
+[here](cylc-8-tasks#general-authentication-issues-sujata-hilary--damian-and-martin-)
+on usage of tokens for authentication between any workflow & the workflow
+service. Both of the following token approaches have been raised as potential
+solutions:
+
+* "one-time" tokens: single-use tokens i.e. ones that are created then deleted
+  once they are no longer required, with fine (process) granularity for
+  creation/deletion;
+* "timed" tokens: tokens that expire after a set time period.
+
+Due e.g. to the new architecture, job clients must be distinguished & treated
+differently from user (UI & interactive CLI) clients, unlike for the
+[simple passphrase-based approach of Cylc 7](cylc-7-architecture#authentication).
```

Comment:
The *passphrase* in the old system is really a suite-level one-time-token as it is generated randomly, and is valid for the lifetime of the suite. It is not a mnemonic passphrase provided by a human user.

Reply:
It can actually be a mnemonic passphrase provided by the user (if you do that, the suite server won't overwrite it) - but no one ever does that!

Reply:
We also talked about "event-expired tokens" in our Exeter meetings. The token would be expired *by the job finishing*.  But presumably this is what you actually mean by "one time tokens" because every job necessarily uses the CLI (for messages and potentially otherwise) multiple separate times (not "one time")?  If so, maybe we should call them **one job tokens**?

Reply:
Yes. A token that is only valid for the duration of a job.

Reply:
> It can actually be a mnemonic passphrase provided by the user (if you do that, the suite server won't overwrite it) - but no one ever does that!

Don't tell anyone!

Reply:
Yes, it seems "token" is the more appropriate word strictly but I stuck with passphrase since it is the standard terminology in the old approach.

> suite-level one-time-token ...  valid for the lifetime of the suite

Excellent way to summarise the old method, I'll add that in as it helps to make the difference in that & the proposed new approach very clear.